Quiz/src/components/Quiz/Quiz.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState } from "react";
+import { Progress, Button, Statistic } from "antd";
+const { Countdown } = Statistic;
+import "./Quiz.css";
+
+type Questions = {
+  questions: Question[];
+};
+type Question = {
+  question: string;
+  choices: string[];
+  type: string;
+  correctAnswer: string;
+};
+
+const Quiz: React.FC<Questions> = ({ questions }) => {
+  const [currentQuestion, setCurrentQuestion] = useState(0);
+  const { question, choices } = questions[currentQuestion];
+  const onChoiceClick = () => {
+    if (currentQuestion !== questions.length - 1) {
+      setCurrentQuestion((prev) => prev + 1);
+    } else {
+      setCurrentQuestion(0);
+    }
+  };
```

Comment:
```suggestion
const onChoiceClick = useCallback(() => {
    setCurrentQuestion(prev => (prev < questions.length - 1) ? prev + 1 : 0);
}, [questions.length, setCurrentQuestion]);
```
the condition can be inside of calling setCurrentQuestion
'cause there is a call of setCurrentQuestion anyway